n_helpers.c

```diff
@@ -74,9 +79,204 @@ static short normalYearDaysByMonth[] = {0, 31, 59, 90, 120, 151, 181, 212, 243,
 static const char *daynames[] = {"Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"};
 
 // Forwards
+NOTE_C_STATIC void setTime(JTIME seconds);
 NOTE_C_STATIC bool timerExpiredSecs(uint32_t *timer, uint32_t periodSecs);
 NOTE_C_STATIC int ytodays(int year);
 
+//**************************************************************************/
+/*!
+  @brief  Receive a large binary object from the Notecard's binary buffer
+  @param  buffer A buffer to hold the result
+  @param  bufLen The total length of the buffer
+  @returns  NULL on success, else an error string pointer.
+  @note  Buffers are decoded in place, the original contents of the buffer will
+         be modified.
+*/
+/**************************************************************************/
+const char * NoteBinaryReceive(uint8_t * buffer, size_t bufLen)
+{
+    // Issue a "card.binary" request.
+    J *rsp = NoteRequestResponse(NoteNewRequest("card.binary"));
+    if (!rsp) {
+        return ERRSTR("unable to issue binary request", c_err);
+    }
+
+    // Ensure the transaction doesn't return an error
+    // to confirm the binary feature is available
+    if (NoteResponseError(rsp)) {
+        JDelete(rsp);
+        return ERRSTR("feature not implemented", c_bad);
```

Comment:
I would rather get the actual error in the response as opposed to "feature not implemented". The feature might be implemented and the actual error could be something else. Not sure if there's an easy way to do that here, though, since the return type is a const string... maybe just log it with `_Debug`?

Reply:
Okay, I'll will inspect the message to eliminate assumptions, as well as issue a `_Debug` message.